source/Calamari.Tests/KubernetesFixtures/ResourceStatus/ResourceStatusReportWrapperTests.cs

```diff
@@ -0,0 +1,240 @@
+    public class ResourceStatusReportWrapperTests
+    {
+        private const ScriptSyntax Syntax = ScriptSyntax.Bash;
+        
+        [Test]
+        public void Enabled_WhenDeployingToAKubernetesClusterWithStatusCheckEnabled()
+        {
+            var variables = new CalamariVariables();
+            var log = new SilentLog();
+            var fileSystem = new TestCalamariPhysicalFileSystem();
+            var statusChecker = new MockResourceStatusChecker();
+
+            variables.Set(Deployment.SpecialVariables.EnabledFeatureToggles, "KubernetesDeploymentStatusFeatureToggle");
+            variables.Set(SpecialVariables.ClusterUrl, "https://localhost");
+            variables.Set(SpecialVariables.ResourceStatusCheck, "True");
+            
+            var wrapper = new ResourceStatusReportWrapper(variables, log, fileSystem, statusChecker);
+
+            wrapper.IsEnabled(Syntax).Should().BeTrue();
+        }
+
+        [Test]
+        public void NotEnabled_WhenDeployingToAKubernetesClusterWithStatusCheckDisabled()
+        {
+            var variables = new CalamariVariables();
+            var log = new SilentLog();
+            var fileSystem = new TestCalamariPhysicalFileSystem();
+            var statusChecker = new MockResourceStatusChecker();
+
+            variables.Set(Deployment.SpecialVariables.EnabledFeatureToggles, "KubernetesDeploymentStatusFeatureToggle");
+            variables.Set(SpecialVariables.ClusterUrl, "https://localhost");
+
+            var wrapper = new ResourceStatusReportWrapper(variables, log, fileSystem, statusChecker);
+
+            wrapper.IsEnabled(Syntax).Should().BeFalse();
+        }
+        
+        [Test]
+        public void NotEnabled_WhenDoingABlueGreenDeployment()
+        {
+            var variables = new CalamariVariables();
+            var log = new SilentLog();
+            var fileSystem = new TestCalamariPhysicalFileSystem();
+            var statusChecker = new MockResourceStatusChecker();
+
+            variables.Set(Deployment.SpecialVariables.EnabledFeatureToggles, "KubernetesDeploymentStatusFeatureToggle");
+            variables.Set(SpecialVariables.ClusterUrl, "https://localhost");
+            variables.Set(SpecialVariables.ResourceStatusCheck, "True");
+            variables.Set(SpecialVariables.DeploymentStyle, "bluegreen");
+            
+            var wrapper = new ResourceStatusReportWrapper(variables, log, fileSystem, statusChecker);
+
+            wrapper.IsEnabled(Syntax).Should().BeFalse();
+        }
+
+        [Test]
+        public void NotEnabled_WhenWaitForDeploymentIsSelected()
+        {
+            var variables = new CalamariVariables();
+            var log = new SilentLog();
+            var fileSystem = new TestCalamariPhysicalFileSystem();
+            var statusChecker = new MockResourceStatusChecker();
+            
+            variables.Set(Deployment.SpecialVariables.EnabledFeatureToggles, "KubernetesDeploymentStatusFeatureToggle");
```

Comment:
i noticed these 3 lines are repeated pretty much on every test, is this something we can do differently, eg `CreateCalamariVariables`